exporter/ethstore.go

```diff
@@ -48,11 +48,22 @@ func StartEthStoreExporter(bnAddress string, enAddress string, updateInterval, e
 		ese.Sleep = time.Minute
 	}
 
+	// Reexport days if specified
+	if startDayReexport != -1 && endDayReexport != -1 {
+		for day := startDayReexport; day <= endDayReexport; day++ {
+			err := ese.exportDay(strconv.FormatInt(day, 10), true)
+			if err != nil {
+				logger.WithError(err).Errorf("error reexporting eth.store day %d in database", day)
+				return
+			}
+		}
+	}
+
 	ese.Run()
```

Comment:
A reexport of certain days most likely should not execute `ese.Run()` too, should it?

Reply:
My logic was that when the ethstore exporter gets started you can give it some optional days to reexport first and then it continues to run as normal.

If I read it right the rewards-exporter does it the same way if the flag **epoch** is -1.

Reply:
I do believe that you misread the sources of the reward exporter. Similar to your exporter, it has two settings to trigger a re-export: `epoch` as well as `epochStart` and `epochEnd`:

---

If `epochEnd` is set
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/d767cf39f0e345fdc21b669f71655102d94de422/cmd/rewards-exporter/main.go#L76
epochs between `epochStart` and `epochEnd` are reexported and the program then ends here:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/d767cf39f0e345fdc21b669f71655102d94de422/cmd/rewards-exporter/main.go#L139

---

If `epoch` is NOT set (as well as `epochEnd` that has been checked above)
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/d767cf39f0e345fdc21b669f71655102d94de422/cmd/rewards-exporter/main.go#L142
an endless loop is started and the missing epochs are reexported periodically.
But if `epoch` is set, only a single epoch is reexported and the program ends:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/d767cf39f0e345fdc21b669f71655102d94de422/cmd/rewards-exporter/main.go#L176-L179

Reply:
You are right, I guess I missed that all important `return`.
In that case I now return after a reexport.